internal/api/handler.go

```diff
@@ -0,0 +1,61 @@
+package api
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/JakeHumphries/gymshark-hacker-news/internal/models"
+	"github.com/labstack/echo/v4"
+	"github.com/pkg/errors"
+)
+
+// ItemReader is an interface for getting items
+type ItemReader interface {
+	GetAllItems(ctx context.Context) ([]models.Item, error)
+	GetStories(ctx context.Context) ([]models.Item, error)
+	GetJobs(ctx context.Context) ([]models.Item, error)
+}
+
+// Handler contains all the business logic for handling requests
+type Handler struct {
+	itemReader ItemReader
+	ctx        context.Context
```

Comment:
Rather than storing `context.Context` in a struct, it's better if you pass it to each function that requires it. [Read more here](https://go.dev/blog/context-and-structs)